src/vtbackend/ViInputHandler.cpp

```diff
@@ -444,25 +457,26 @@ void ViInputHandler::startSearchExternally()
 
 Handled ViInputHandler::handleSearchEditor(char32_t ch, Modifiers modifiers)
 {
-    assert(_searchEditMode != SearchEditMode::Disabled);
+    assert(_searchEditMode != PromptMode::Disabled);
```

Comment:
let's see if we can find a way to not duplicate the input handling by "dependency injection", implementing the custom behavior through some common interface (search vs tab naming vs future-feature)